components/scream/src/share/atm_process/atmosphere_process.cpp

```diff
@@ -818,4 +824,37 @@ ::remove_group (const std::string& group_name, const std::string& grid_name) {
   rmg(m_groups_out, m_groups_out_pointers);
 }
 
+void AtmosphereProcess::setup_conservation_checks_for_this_process (const int dt)
+{
+  // Only setup the check if this process call it.
+  if (not m_params.get<bool>("enable_column_conservation_checks", false)) {
+    return;
+  }
+
+  // Tolerance for the relative error.
+  const Real tol = m_params.get<Real>("column_conservation_check_tolerance",
+                                      1e-10);
+
+  // Loop through postcondition checks, find the MassAndEnergyConservationCheck,
+  // set dt and tolerance, and compute current mass and energy.
+  bool conservation_check_exists = false;
+  for (const auto& it : m_postcondition_checks) {
+    const auto& pc = it.second;
+    if (pc->name() == "Energy conservation check") {
+      conservation_check_exists = true;
+
+      const auto conservation_check = std::dynamic_pointer_cast<MassAndEnergyConservationCheck>(pc);
+      conservation_check->set_dt(dt);
```

Comment:
If I understand correctly, you've resolved the dt issue by setting it each time the process is called in the run_impl loop. The dt passed here is the one the run_impl function is given, and that value is dtime divided by the proper number of subcycles. Right?

Reply:
Correct. So the dt passed to the checker is `atm_dt/num_current_subcycles`